src/app/shared/components/template/services/instance/template-instance.service.ts

```diff
@@ -0,0 +1,10 @@
+import { Injectable, Injector, ProviderToken } from "@angular/core";
+
+@Injectable({ providedIn: "root" })
+export class TemplateInstanceService {
+  constructor(public injector: Injector) {}
+
+  getGlobalService<T>(token: ProviderToken<T>) {
+    return this.injector.get(token);
+  }
+}
```

Comment:
This is the common method used by instance services. The `injector` provides a link to angular's system for handling dependency injection, and is used to get reference to existing global services